build.sbt

```diff
@@ -197,58 +204,87 @@ lazy val macrosAggregate = aggregateProject("macros", macrosJS, macrosJVM)
 lazy val macros = crossProject(JSPlatform, JVMPlatform)
   .crossType(CrossType.Pure)
   .in(file("macros"))
-  .settings(testSettings: _*)
-  .jsSettings(jsTestSettings: _*)
-  .settings(commonWithPublishSettings: _*)
-  .settings(withCompatUnmanagedSources(jsJvmCrossProject = true, includeTestSrcs = false): _*)
+  .settings(testSettings)
+  .jsSettings(jsTestSettings)
+  .settings(commonWithPublishSettings)
+  .settings(withCompatUnmanagedSources(jsJvmCrossProject = true, includeTestSrcs = false))
   .settings(
     name    := "enumeratum-macros",
     version := Versions.Macros.head,
     crossScalaVersions := scalaVersionsAll, // eventually move this to aggregateProject once more 2.13 libs are out
-    libraryDependencies ++= Seq(
-      "org.scala-lang" % "scala-reflect" % scalaVersion.value
-    )
+    libraryDependencies ++= {
+      if (scalaBinaryVersion.value startsWith "2.") {
+        Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value)
+      } else {
+        Seq(
+          "org.scala-lang" %% "scala3-compiler" % scalaVersion.value % Provided
+        )
+      }
+    }
   )
 
 lazy val macrosJS  = macros.js
 lazy val macrosJVM = macros.jvm
 
+lazy val useLocalVersion = sys.props.get("enumeratum.useLocalVersion").nonEmpty
+
 // Aggregates core
 lazy val coreAggregate = aggregateProject("core", coreJS, coreJVM)
 lazy val core = crossProject(JSPlatform, JVMPlatform)
   .crossType(CrossType.Pure)
   .in(file("enumeratum-core"))
-  .settings(testSettings: _*)
-  .jsSettings(jsTestSettings: _*)
-  .settings(commonWithPublishSettings: _*)
+  .settings(testSettings)
+  .jsSettings(jsTestSettings)
+  .settings(commonWithPublishSettings)
   .settings(
-    name                                  := "enumeratum",
-    version                               := Versions.Core.head,
-    crossScalaVersions                    := scalaVersionsAll,
-    libraryDependencies += "com.beachape" %% "enumeratum-macros" % Versions.Macros.stable
+    name               := "enumeratum",
+    version            := Versions.Core.head,
+    crossScalaVersions := scalaVersionsAll,
+    libraryDependencies ++= {
+      if (useLocalVersion) {
+        Seq.empty
+      } else {
+        Seq("com.beachape" %% "enumeratum-macros" % Versions.Macros.stable)
+      }
+    }
   )
-// .dependsOn(macros) // used for testing macros
-lazy val coreJS  = core.js
-lazy val coreJVM = core.jvm
+
+def configureWithLocal(m: Project): Project => Project = { p =>
```

Comment:
Utility to use local project/dependency rather than stable/previous version dependency